javascript/ql/lib/semmle/javascript/frameworks/WebSocket.qll

```diff
@@ -214,7 +214,13 @@ module ServerWebSocket {
   class ServerSocket extends EventEmitter::Range, DataFlow::SourceNode {
     LibraryName library;
 
-    ServerSocket() { this = getAConnectionCall(library).getCallback(1).getParameter(0) }
+    ServerSocket() {
+      this = getAConnectionCall(library).getCallback(1).getParameter(0)
```

Comment:
Out of curiosity what is the difference between `getCallback(1).getParameter(0)` and `getABoundCallbackParameter(1, 0)` ?

Reply:
The latter transparently handles partial function calls using `Function.prototype.bind`.  

E.g.  
```JavaScript
function myRawCB(bound, free) {
  // ....
}
// bind the first argument to "bound"
var myCB = myRawCB.bind(null, "bound");

library(foo, myCB); // `free` is the zeroth argument in the resulting callback. So reachable with `getABoundCallbackParameter(1, 0)`
```